(feat): Lambda response streaming docs (#20299)
newrelic/docs-website: src/content/docs/serverless-function-monitoring/aws-lambda-monitoring/response-stream.mdx

---
title: Monitor response streaming Lambda functions

> metaDescription: "Learn how to monitor AWS Lambda functions that utilize response streaming in the New Relic."

Reviewer: ```suggestion
metaDescription: "Learn how to monitor AWS Lambda functions that use response streaming in the New Relic."
```

freshnessValidatedDate: never
tags:
    - aws
    - lambda
    - response stream monitoring
    - ai monitoring
---

> You can monitor AWS Lambda functions that use [response streaming](https://aws.amazon.com/blogs/compute/introducing-aws-lambda-response-streaming/) in Node.js applications to gain real-time insights into the performance and behavior of each invocation. This solution integrates with [AI monitoring](/docs/ai-monitoring/intro-to-ai-monitoring/) to allow you to track model responses and collect observability data related to Large Language Models (LLMs) in your AI workloads. By leveraging these capabilities, you can optimize your serverless applications' performance and maintain comprehensive oversight of AI-driven processes.

Reviewer: ```suggestion
You can monitor AWS Lambda functions that use [response streaming](https://aws.amazon.com/blogs/compute/introducing-aws-lambda-response-streaming/) in Node.js applications to get real-time insights into the performance and behavior of each invocation. This solution integrates with [AI monitoring](/docs/ai-monitoring/intro-to-ai-monitoring/) to allow you to track model responses and collect observability data related to Large Language Models (LLMs) in your AI workloads. Using these capabilities, you can optimize your serverless applications' performance and maintain comprehensive oversight of AI-driven processes.
```

This document helps you to:

* Set up Lambda response streaming monitoring.
* Understand the metrics and data available for monitoring Lambda response streaming.
* Learn how to use the New Relic UI to monitor and observe the performance of your Lambda response streaming.

New image needed!

<img
    title="New Relic AI Responses page"
    alt="Lambda response monitoring in the New Relic platform"
    src="/images/ss-ai-response-page.gif"
/>

## Set up the Lambda response streaming

### Prerequisites [#prerequisites]

> * Check that your Lambda function meets our [compatibility and requirements](/docs/serverless-function-monitoring/aws-lambda-monitoring/instrument-lambda-function/compatibility-requirement-lambda-monitoring).

Reviewer: ```suggestion
* Check that your Lambda function meets our [compatibility requirements](/docs/serverless-function-monitoring/aws-lambda-monitoring/instrument-lambda-function/compatibility-requirement-lambda-monitoring).
```

### Configure the Lambda response streaming [#configure-response-streaming]

To enable response streaming monitoring for your Node.js Lambda functions:

1. Install and configure [Lambda monitoring on New Relic](/docs/serverless-function-monitoring/aws-lambda-monitoring/instrument-lambda-function/compatibility-requirement-lambda-monitoring).

2. Add the required [environment variables](/docs/serverless-function-monitoring/aws-lambda-monitoring/instrument-lambda-function/env-variables-lambda) for node.js response streaming Lambda functions.

3. Save the changes and deploy the function.

>     After deployment, you can view the Lambda response streaming data in the New Relic platform. For more information, See [View the Lambda response streaming data](#view-response-streaming-data).

Reviewer: ```suggestion
    After deployment, you can view the Lambda response streaming data in the New Relic platform. For more information, refer [View the Lambda response streaming data](#view-response-streaming-data).
```

## Access the Lambda response streaming data [#access-ui]

To view the Lambda response streaming data in the New Relic platform:

> 1. Log in to your New Relic account.

Reviewer: This step is not necessary to call out. It is implicit.

2. Go to the **left navigation pane > All Capabilities > Serverless Functions**.

> 3. Select the required Lambda function that utilizes response streaming.

Reviewer: ```suggestion
3. Select the required Lambda function that uses response streaming.
```

4. In the **More views** section, click **AI Responses**.

<img
    title="New Relic AI Responses page"
    alt="Lambda response monitoring in the New Relic platform"
    src="/images/ss-ai-response-page.gif"
/>

>     To know more about AI responses, see [view AI responses](/docs/ai-monitoring/explore-ai-data/view-ai-responses).

Reviewer: ```suggestion
    To know more about AI responses, refer [view AI responses](/docs/ai-monitoring/explore-ai-data/view-ai-responses).
```

>     To know more about AI responses, see [view AI responses](/docs/ai-monitoring/explore-ai-data/view-ai-responses).

Reviewer: Is "view AI responses" the title of the article? If so, follow correct capitalization
